oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/TraceQueryService.java

```diff
@@ -244,4 +268,61 @@ private void findChildren(List<Span> spans, Span parentSpan, List<Span> children
             }
         });
     }
+
+    private void appendAttachedEventsToSpan(List<Span> spans, List<SpanAttachedEventRecord> events) throws InvalidProtocolBufferException {
+        if (CollectionUtils.isEmpty(events)) {
+            return;
+        }
+
+        for (SpanAttachedEventRecord record : events) {
+            if (!StringUtils.isNumeric(record.getTraceSpanId())) {
+                continue;
+            }
+
+            // find matches span
+            final int eventSpanId = Integer.parseInt(record.getTraceSpanId());
```

Comment:
I think we should check SpanReferenceType first, then do casting. Otherwise, we may face NPE or cast error, if trace ID actually exists on both sides. Such as Lua agent uses UUID as well.

Reply:
According to this, https://github.com/apache/skywalking-query-protocol/pull/101, we should update the Zipkin query as well. And verify whether we could put this in tags or some places for lens UI?

Reply:
Updated. I have filtered the skywalking events in the query and checked the parent span ID is numeric or not.